perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -1729,7 +1729,7 @@ public RichUser convertUserToRichUserWithAttributesByNames(PerunSession sess, Us
 	public List<RichUser> findRichUsersWithAttributes(PerunSession sess, String searchString, List<String> attrsName) throws UserNotExistsException {
 
 		if(attrsName == null || attrsName.isEmpty()) {
-			return convertRichUsersToRichUsersWithAttributes(sess, findRichUsers(sess, searchString));
+			return findRichUsers(sess, searchString);
```

Comment:
Same optimization can be done for ``findRichUsersWithAttributesByExactMatch()`` below.

Also I wonder, if we can optimize implementation of ``convertRichUsersToRichUsersWithAttributes()`` itself, which retrieves user by id. We can easily pass ``RichUser`` instead of ``User`` into ``getAttributes()`` method, but I'm not sure that some code doesn't rely on the exact class to be used.

Reply:
> Same optimization can be done for `findRichUsersWithAttributesByExactMatch()` below.

Done.

> Also I wonder, if we can optimize implementation of `convertRichUsersToRichUsersWithAttributes()` itself, which retrieves user by id. We can easily pass `RichUser` instead of `User` into `getAttributes()` method, but I'm not sure that some code doesn't rely on the exact class to be used.

I have noticed this as well. I took a look at it and I think this should not cause any problems. So I have added also this optimization.

Reply:
Done.